app/plugin_api.go

```diff
@@ -650,3 +650,33 @@ func (api *PluginAPI) LogError(msg string, keyValuePairs ...interface{}) {
 func (api *PluginAPI) LogWarn(msg string, keyValuePairs ...interface{}) {
 	api.logger.Warn(msg, keyValuePairs...)
 }
+
+func (api *PluginAPI) CreateBot(bot *model.Bot) (*model.Bot, *model.AppError) {
+	// Bots created by a plugin should use the plugin's ID for the creator field, unless 
+	// otherwise specified by the plugin.
+	if bot.CreatorId == "" {
+		bot.CreatorId = api.id
+	}
+
+	return api.app.CreateBot(bot)
+}
+
+func (api *PluginAPI) PatchBot(userId string, botPatch *model.BotPatch) (*model.Bot, *model.AppError) {
+	return api.app.PatchBot(userId, botPatch)
+}
+
+func (api *PluginAPI) GetBot(userId string, includeDeleted bool) (*model.Bot, *model.AppError) {
+	return api.app.GetBot(userId, includeDeleted)
+}
+
```

Comment:
What do you think about adding a `GetBotByUsername` method? This would be handy as plugins won't have to store the `userId` from there own bots.

Reply:
I'm a proponent of keeping the plugin API surface as small as possible, introducing new methods only to 1) expose new functionality, 2) address performance concerns.

A `GetBotByUsername` is definitely useful, but I'm thinking it fits better into a yet-to-become-available layer on top of the plugin API that is versioned independently from the mattermost-server. I'm thinking of something like a `mattermost-plugin-extended-api` that is imported by plugins and for which we go nuts on helper methods and whatever, but keep the core RPC interface "boring".

Reply:
I'm full onboard about not blowing up the API.

But I would argue that `GetBotByUsername` is at least equally important then `GetBot`. I suppose a typical workflow of a plugin is:
- On activation the plugin checks if "there" bot exists.
- If not, it creates the bot.
- It creates posts with this bot.

This is trivial to do with `GetBotByUsername` because the plugins knows the Username of there Bot. But without this method, this common task becomes more difficult. It would have to get the bot via `GetBots`, but only if it knows the `GetBots`. Or the plugin can store the `userID`, but this would be additional complexity.

Reply:
Discussed offline: Plugin devs can use `GetUserByUsername()` for this. We try to keep the API small for now.